Fix race condition between http error notification and server-side validation error notification (#9848)
marmelab/react-admin: docs/Validation.md

Server-side validation is supported out of the box for `pessimistic` mode only. It requires that the dataProvider throws an error with the following shape:

```json
{
    "body": {
        "errors": {
            // Global validation error message (optional)
            "root": { "serverError": "Some of the provided values are not valid. Please fix them and retry." },
            // Field validation error messages
            "title": "An article with this title already exists. The title must be unique.",
            "date": "The date is required",
            "tags": { "message": "The tag 'agrriculture' doesn't exist" },
        }
    }
}
```

**Tip**: The shape of the returned validation errors must match the form shape: each key needs to match a `source` prop. The only exception is the `root.serverError` key, which can be used to define a global error message for the form.

> **Tip**: The returned validation errors might have any validation format we support (simple strings, translation strings or translation objects with a `message` attribute) for each key. However `root.serverError` must be a plain string.

Reviewer: This is confusing. is a translation string a plain string? As we pass the error to `notify`, translation is supported, isn't it?
```suggestion
**Tip**: The returned validation errors might have any validation format we support (simple strings, translation strings or translation objects with a `message` attribute) for each key. However `root.serverError` does not accept translation objects.
```
